src/others/observer/lv_observer.c

```diff
@@ -393,6 +388,11 @@ void lv_observer_remove(lv_observer_t * observer)
 {
     LV_ASSERT_NULL(observer);
 
+    if(observer->for_obj && observer->target) {
+        lv_obj_remove_event_cb(observer->target, unsubscribe_on_delete_cb);
```

Comment:
To make sure that the correct event is removed if the object is subscribed to multiple subjects: 
```suggestion
        lv_obj_remove_event_cb_with_user_data(observer->target, unsubscribe_on_delete_cb, observer);
```

Reply:
Actually, seeing this now, I realized that is vital because it isolates the disconnect ONLY from that one observer, whereas that function is shared among all possible observers connected with that Widget!  Good catch, @kisvegabor !!